ballista/rust/core/Cargo.toml

```diff
@@ -33,6 +33,7 @@ simd = ["datafusion/simd"]
 ahash = "0.7"
 async-trait = "0.1.36"
 futures = "0.3"
+hashbrown = "0.11"
```

Comment:
From the official readme:

```
Since Rust 1.36, this is now the HashMap implementation for the Rust standard library. However you may still want to use this crate instead since it works in environments without std, such as embedded systems and kernels.
```

Are we introducing it because we are expecting to use ballista in embedded systems?

Reply:
It looks like the dependency was added to DataFusion for a performance optimization in https://github.com/apache/arrow-datafusion/commit/ec062075cd03fc68e22b96d6f8bf0ed291e30d93 and when the `metrics` method was added in DataFusion is used this hashmap as well, but perhaps we should have had it use the std hashmap instead?

@Dandandan what do you think?

Reply:
Filed https://github.com/apache/arrow-datafusion/issues/677 as a follow up so we don't block the merge of this PR. I am curious whether the std hashmap also uses ahash as the default hasher.

Reply:
Added some comments on the ticket.